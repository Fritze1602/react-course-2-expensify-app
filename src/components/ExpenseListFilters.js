import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates'

class ExpenseListFilters extends React.Component{
    state = {
        calenderFocused: null
    }
    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate))
    }
    onFocusChange = (calenderFocused) => {
        this.setState(()=>({
            calenderFocused
        }))
    }
    render () {
        return (
            <div>
                <input 
                    type="text" 
                    value={this.props.filters.text} 
                    onChange={(e)=>{
                        this.props.dispatch(setTextFilter(e.target.value));
                }}/>
                <select 
                    value={this.props.filters.sortBy}
                    onChange={(e) => {
                        e.target.value === 'amount' ? this.props.dispatch(sortByAmount()) : this.props.dispatch(sortByDate());
                }}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    startDateId="your_unique_start_date_id" 
                    endDate={this.props.filters.endDate}
                    endDateId="your_unique_end_date_id" 
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calenderFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={()=>false}
                    minimumNights={0}

                />
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
        filters:state.filters
    }
}

export default connect(mapStateToProps)(ExpenseListFilters);