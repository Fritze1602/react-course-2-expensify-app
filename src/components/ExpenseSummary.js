import React from 'react';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral'
import {connect} from 'react-redux';

export const ExpenseSummary = ({expensesCount, expensesTotal}) => {
    const expensesWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00');
    return (
        <div>
            <p>Viewing {expensesCount} {expensesWord} with amount of {formattedExpensesTotal}.</p>
        </div>
    )
}

const mapStateToProps = (state)=>{
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary);

