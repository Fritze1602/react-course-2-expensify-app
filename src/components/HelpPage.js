import React from 'react';

const HelpPage = () => (
    <div>This is my new Help Page</div>
);

export default HelpPage;