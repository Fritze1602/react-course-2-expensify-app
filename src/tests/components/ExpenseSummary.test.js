import React from 'react';
import {shallow} from 'enzyme';

import {ExpenseSummary} from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';


test('Should render Summary for multiple specific expense', () => {
    const wrapper = shallow( 
    <ExpenseSummary 
        expensesCount = {4}
        expensesTotal = {278}  /> );
    expect(wrapper).toMatchSnapshot();
})

test('Should render Summary for one expense', () => {
    const wrapper = shallow( <ExpenseSummary 
        expensesCount = {1} 
        expensesTotal = {2555}  /> );
    expect(wrapper).toMatchSnapshot();
})

