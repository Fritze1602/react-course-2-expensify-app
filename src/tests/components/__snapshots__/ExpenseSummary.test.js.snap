// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Summary for multiple specific expense 1`] = `
<div>
  <p>
    Viewing 
    4
     
    expenses
     with amount of 
    $2.78
    .
  </p>
</div>
`;

exports[`Should render Summary for one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     with amount of 
    $25.55
    .
  </p>
</div>
`;
